src/components/CromwellModal.js

```diff
@@ -0,0 +1,88 @@
+import _ from 'lodash/fp'
+import { Fragment, useState } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { ButtonPrimary, spinnerOverlay } from 'src/components/common'
+import { withModalDrawer } from 'src/components/ModalDrawer'
+import { tools } from 'src/components/notebook-utils'
+import TitleBar from 'src/components/TitleBar'
+import { Ajax } from 'src/libs/ajax'
+import { withErrorReporting } from 'src/libs/error'
+import Events, { extractWorkspaceDetails } from 'src/libs/events'
+import {
+  computeStyles,
+  getCurrentAppForType
+} from 'src/libs/runtime-utils'
+import * as Utils from 'src/libs/utils'
+
+
+const defaultDataDiskSize = 500 // GB
+const defaultKubernetesRuntimeConfig = { machineType: 'n1-highmem-8', numNodes: 1, autoscalingEnabled: false }
+
+const titleId = 'cromwell-modal-title'
+
+export const CromwellModalBase = Utils.withDisplayName('CromwellModal')(
+  ({
+    onDismiss, onSuccess, apps, appDataDisks, workspace, workspace: { workspace: { namespace, bucketName, name: workspaceName, googleProject } },
+    isAnalysisMode = false
+  }) => {
+    const app = getCurrentAppForType(tools.cromwell.appType)(apps)
+    const [loading, setLoading] = useState(false)
+
+    const createCromwell = _.flow(
+      Utils.withBusyState(setLoading),
+      withErrorReporting('Error creating Cromwell')
+    )(async () => {
+      await Ajax().Apps.app(googleProject, Utils.generateAppName()).create({
+        defaultKubernetesRuntimeConfig, diskName: Utils.generatePersistentDiskName(), diskSize: defaultDataDiskSize,
+        appType: tools.cromwell.appType, namespace, bucketName, workspaceName
+      })
+      Ajax().Metrics.captureEvent(Events.applicationCreate, { app: tools.cromwell.appType, ...extractWorkspaceDetails(workspace) })
+      return onSuccess()
+    })
+
+    const renderActionButton = () => {
+      return !app ? h(ButtonPrimary, { onClick: createCromwell }, ['Create']) :
+        h(ButtonPrimary, { onClick: onDismiss }, ['Close'])
```

Comment:
should there be a delete button if there is a running app, or is that out of scope?

Reply:
It's out of scope at this point- you have to go to the Cloud Environments page to delete Cromwell at the moment. This ticket was really just about starting Cromwell (and we want to work with UX to design what we display in the "Cromwell modal").